### PR TITLE
Dev/57241 improve noise options

### PR DESCRIPTION
**Pull request by elisemicho**
1. space insertion (we already have drop)
2. char substitution with neighbor letters on qwerty keyboard (we already have drop, swap and duplicate)
3. possibility to tag noisy source data

### CODE REVIEW

File: nmtwizard/preprocess/operators/noise.py [hunk 1]
```diff
@@ -41,17 +45,49 @@ def __init__(self, config, *args):
...
 
     def _preprocess_tu(self, tu, *args):
+        tu = self._apply_space_insertion_noise(tu)
         src_tok = tu.src_tok
         tokens = src_tok.token_objects
+        original_tokens = copy.deepcopy(tokens)
         new_tokens = [self._apply_word_noise(tokens[0])]
         tu.src_tok = (src_tok.tokenizer, new_tokens)
+        if self._add_marker and new_tokens != original_tokens:
+            tu.replace_tokens_side("source", (0, 0, ["｟mrk_noisy｠"]))
```
Comment (natsegal) [inline nmtwizard/preprocess/operators/noise.py:63]: Looks like this check doesn't account for space insertion. Is it expected ?

Comment (elisemicho) [reply nmtwizard/preprocess/operators/noise.py:63]: No, it was not expected, thanks for seeing that!

**Comment by natsegal** [comment]
Looks good to me. @guillaumekln, do you have any comments/suggestions ?

**Review by guillaumekln — APPROVED** [review]
Looks good to me.

**Pull request merged**